src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/BodyModelBinder.cs

```diff
@@ -177,9 +177,26 @@ public BodyModelBinder(IList<IInputFormatter> formatters, IHttpRequestStreamRead
                     bindingContext.ModelState.AddModelError(modelBindingKey, message);
                 }
             }
-            catch (Exception ex)
+            catch (InputFormatterException InputFormatterException)
             {
-                bindingContext.ModelState.AddModelError(modelBindingKey, ex, bindingContext.ModelMetadata);
+                bindingContext.ModelState.AddModelError(modelBindingKey, InputFormatterException, bindingContext.ModelMetadata);
+            }
+            catch (Exception exception)
+            {
+                var sendBadRequestForAllExceptions = _options.ConvertAllExceptionsToModelStateErrors;
+                if (formatter is IFormatterExceptionPolicy exceptionPolicy && exceptionPolicy.HandlesExceptions)
+                {
+                    sendBadRequestForAllExceptions = _options.ConvertAllExceptionsOfInbuiltInputFormattersToModelStateErrors;
+                }
+
+                if (sendBadRequestForAllExceptions)
```

Comment:
Can we use different terminology here? Nothing about input formatters "sends bad request". They either handle exceptions or let them propagate